docs/notebooks/rembo.pct.py

```diff
@@ -0,0 +1,246 @@
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this example, we augment the standard two-dimensional Michalewicz function with 98 dummy dimensions to obtain a 100-dimensional problem over the hypercube $[0, \pi]^{100}$.
+#
+# We compare three approaches to optimizing this problem. The first uses a standard GP model over all 100 dimensions, using expected improvement as our acquisition function. We compare this to two Random EMbedding Bayesian Optimization (REMBO; see <cite data-cite="wang2013bayesian"/>) approaches.
+#
+# Instead of training a GP model and optimizing an acquisition function on the high-dimensional space directly, REMBO constructs a low-dimensional search space, performing the modeling and acquisition on this space. In order to transfer to the high-dimensional space, REMBO uses a static random projection matrix $A \in \mathbb{R}^{D \times d}$ to project query points from the lower, $d$-dimensional space to the original higher, $D$-dimensional space.
+#
+# As the lower dimension $d$ is a choice, we compare $d = 2$ and $d = 5$. While $d = 2$ should be sufficient (as the problem is intrinsically two-dimensional), a higher dimension may improve the chance of a good random embedding being found, at the cost of making it more difficult to find good areas of the lower-dimensional search space.
+#
+# We run each method 5 times to ensure that the results are not due to luck.
+
+# %%
+from trieste.objectives.single_objectives import Michalewicz2
+from trieste.space import Box
+from trieste.models.gpflow import GaussianProcessRegression
+
+D = 100  # the dimension of the full problem
+
+num_initial_points = 2
+num_steps = 48
+num_seeds = 5
+
+objective = Michalewicz2.objective
+minimum = Michalewicz2.minimum
+search_space = (
+    Box([0.0], [math.pi]) ** D
+)  # manually construct the high-dimensional search space
+
+
+# We simply add dummy dimensions to create the new objective
+def high_dim_objective(x):
+    tf.debugging.assert_shapes([(x, (..., D))])
+
+    return objective(x[..., :2])
+
+
+def build_model(data, d):
+    # add a bit of noise, since there's a risk the variance could be zero for Michalewicz
+    variance = tf.math.reduce_variance(data.observations) + 1e-4
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2] * d)
+    prior_scale = tf.cast(1.0, dtype=tf.float64)
+    kernel.variance.prior = tfp.distributions.LogNormal(
+        tf.cast(-2.0, dtype=tf.float64), prior_scale
+    )
+    kernel.lengthscales.prior = tfp.distributions.LogNormal(
+        tf.math.log(kernel.lengthscales), prior_scale
+    )
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GaussianProcessRegression(gpr, num_kernel_samples=100)
+
+
+# %% [markdown]
+# ## Run standard Bayesian optimization
+# We run the process 5 times - note that this takes a while!
```

Comment:
How long does this take?

Regardless, it's probably worth using quickrun to speed it up for the pre-merge quality checks. The idea is to write a regex replace to apply to the file before running it that speeds it up and still catches crashes even if it doesn't generate the right results and graphs. At the very least you should be able to reduce num_steps to something minimal like 2. Feel free to ping me if you're not sure how to do this.

Reply:
I've had a go at this now